src/classes/ALLO_Allocations_TDTM.cls

```diff
@@ -643,7 +643,9 @@ public class ALLO_Allocations_TDTM extends TDTM_Runnable {
                     // At this point, only the amounts on each allocation is correct, not the percentage.
                     // Go through the allocation and adjust the percentage so it is correct.
                     for (Allocation__c consolidatedAllocation : consolidatedAllocationList) {
-                        consolidatedAllocation.Percent__c = consolidatedAllocation.Amount__c / amount * 100;
+                        if(consolidatedAllocation.Percent__c == null){
```

Comment:
Hmmm.  Seems we broke one thing to fix another.  If I remember right, @KietTruongSFDO added this to address a situation where there were more than one of the same allocation from the parent.  Basically, in that scenario, we have to recalculate the % because we are combining two or more allocations from the parent.  It is not setup to differentiate when it does/does not need to do that, though.

Did you play around at all with setting the scale on the percentage?  Possibly to a scale of 0 (since we are multiplying by 100) to get to a whole number.

Reply:
That's what I was afraid of, thanks for the context. I'll add a test case for that situation and update the logic to cover it.

Reply:
@jbrandolinisf I put in some additional logic that will add up the allocation percentages when the same GAU is being used, and only recalculate the percentage if none was provided. I think that should cover our use cases?